src/v/cloud_storage_clients/test_client/s3_test_client_main.cc

```diff
@@ -76,6 +77,11 @@ void cli_opts(boost::program_options::options_description_easy_init opt) {
       po::value<std::string>()->default_value("us-east-1"),
       "aws region");
 
+    opt(
+      "url_style",
+      po::value<std::string>()->default_value("virtual_host"),
```

Comment:
This is good for development but we also need a proper unit-test. In the `src/v/cloud_storage/remote.h` we have a `remote` class which serves as a proxy for the code that has to interact with the cloud storage. All code paths which are downloading or uploading or deleting are using this class instead of the client. The unit-test has to show that when the path-style is enabled no REST API request uses virtual host style URL. The `cloud_storage::remote` has a unit-test suite. It invokes every method of the `remote` and checks the results using the `s3_imposter`. The imposter allows you to examine parameters of the request and also validate the URL. So basically, we need to have all these tests to be invoked with virtual host style URLs and also with path-style URLs.

The motivation here is that some cloud storage providers may not support one style or the other. In this situation we don't want to accidentally use the wrong style.

Reply:
Looks like all existing REST API's that we're using are supported. But we can add new request in the future so some check is nice to have.

Reply:
Could be done in a followup.

Reply:
> Could be done in a followup.

@Lazin Yes, and we are also going to wire up ducktape wiht minio running in both modes (pretty sure it supports it). You can see all teh follow up work in CORE-725.

@WillemKauf can you capture Evgeny's suggestions about additional unit testing in a new or existing jira ticket as a subtask for the overall project?